File: java/src/main/java/org/oojs/model/Schema.java
package org.oojs.model;

import java.util.LinkedHashMap;
import java.util.Map;

/** An OOJS schema document (one per $id). */
public final class Schema {
    public final String oojsVersion;
    public final String schemaId;
    public final String title;
    public final String description;
    public final String discriminator;
    public final boolean closedWorld;

    public Map<String, String> imports = new LinkedHashMap<>();
    public Map<String, TypeDef> types = new LinkedHashMap<>();

    public Schema(String oojsVersion, String schemaId, String title, String description,
            String discriminator, boolean closedWorld) {
        this.oojsVersion = oojsVersion;
        this.schemaId = schemaId;
        this.title = title;
        this.description = description;
        this.discriminator = discriminator;
        this.closedWorld = closedWorld;
    }
}

File: csharp/Oojs/Schema.cs
namespace Oojs;

public sealed class Schema
{
    public Schema(
        string oojsVersion,
        string schemaId,
        string title = "",
        string description = "",
        string discriminator = "_type",
        bool closedWorld = true)
    {
        OojsVersion = oojsVersion;
        SchemaId = schemaId;
        Title = title;
        Description = description;
        Discriminator = discriminator;
        ClosedWorld = closedWorld;
    }

    public string OojsVersion { get; }
    public string SchemaId { get; }
    public string Title { get; }
    public string Description { get; }
    public string Discriminator { get; }
    public bool ClosedWorld { get; }

    public Dictionary<string, string> Imports { get; } = new();
    public Dictionary<string, TypeDef> Types { get; } = new();
}
